src/app/clusters/laundry-washer-controls-server/laundry-washer-controls-server.cpp

```diff
@@ -0,0 +1,133 @@
+
+private:
+    CHIP_ERROR ReadSpinSpeeds(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadSupportedRinses(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder);
+};
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::ReadSpinSpeeds(const ConcreteReadAttributePath & aPath,
+                                                           AttributeValueEncoder & aEncoder)
+{
+    const LaundryWasherControls::LaundryWasherManager * gLaundryWasherManager = LaundryWasherControls::getLaundryWasherManager();
+    const LaundryWasherControls::LaundryWasherManager::AttributeProvider<CharSpan> attrProvider =
+        gLaundryWasherManager->getSpinSpeedProvider(aPath.mEndpointId);
+    if (attrProvider.begin() == nullptr)
+    {
+        aEncoder.EncodeEmptyList();
+        return CHIP_NO_ERROR;
+    }
+    CHIP_ERROR err;
+    err = aEncoder.EncodeList([attrProvider](const auto & encoder) -> CHIP_ERROR {
+        const auto * end = attrProvider.end();
+        for (auto * it = attrProvider.begin(); it != end; ++it)
+        {
+            auto & spinSpeed = *it;
+            ReturnErrorOnFailure(encoder.Encode(spinSpeed));
+        }
+        return CHIP_NO_ERROR;
+    });
+    return err;
+}
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::ReadSupportedRinses(const ConcreteReadAttributePath & aPath,
+                                                                AttributeValueEncoder & aEncoder)
+{
+    const LaundryWasherControls::LaundryWasherManager * gLaundryWasherManager = LaundryWasherControls::getLaundryWasherManager();
+    const LaundryWasherControls::LaundryWasherManager::AttributeProvider<NumberOfRinsesEnum> attrProvider =
+        gLaundryWasherManager->getSupportedRinseProvider(aPath.mEndpointId);
+    if (attrProvider.begin() == nullptr)
+    {
+        aEncoder.EncodeEmptyList();
+        return CHIP_NO_ERROR;
+    }
+    CHIP_ERROR err;
+    err = aEncoder.EncodeList([attrProvider](const auto & encoder) -> CHIP_ERROR {
+        const auto * end = attrProvider.end();
+        for (auto * it = attrProvider.begin(); it != end; ++it)
+        {
+            auto & rinse = *it;
+            ReturnErrorOnFailure(encoder.Encode(rinse));
+        }
+        return CHIP_NO_ERROR;
+    });
+    return err;
+}
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    if (aPath.mClusterId != LaundryWasherControls::Id)
+    {
+        // We shouldn't have been called at all.
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::SpinSpeeds::Id:
+        return ReadSpinSpeeds(aPath, aEncoder);
+    case Attributes::SupportedRinses::Id:
+        return ReadSupportedRinses(aPath, aEncoder);
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+} // namespace
+
+void emberAfLaundryWasherControlsClusterInitCallback(chip::EndpointId endpoint)
```

Comment:
This callback is for application-level setup, no?  Implementing it here means applications can't implement it...

Reply:
I will remove it